src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -852,6 +861,14 @@ public string GetImportedClassName(string className)
 			return collectionMetadata;
 		}
 
+		private void CheckNotDisposed()
+		{
+			if (disposed)
+			{
+				throw new ObjectDisposedException($"Session factory {Name} with id {Uuid}");
+			}
+		}
```

Comment:
I am going to changed this to `CheckNotClosed`, because we can close a factory without disposing it, and closing it is enough to cause the reported troubles.